### PR TITLE
Sending a notification when saving the refresh token to keychain

### PR DESCRIPTION
**Pull request by RLEE0731**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @RLEE0731 has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m [hunk 1]
```diff
@@ -355,6 +361,11 @@ - (void)writeToKeychain
         }
         result = SecItemAdd((CFDictionaryRef)query, NULL);
         BOXAssert( result == noErr, @"Couldn't add the Keychain Item with error %d.", (int)result);
+        NSDictionary *userInfo = @{@"completion_status" : (result == noErr) ? @"succeeded" : @"failed",
+                                   @"return_code" : @(result)};
```
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m]: [@(number) stringValue]

File: BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m [hunk 2]
```diff
@@ -342,6 +343,11 @@ - (void)writeToKeychain
         
         result = SecItemUpdate((CFDictionaryRef)updateItem, (CFDictionaryRef)tempCheck);
         BOXAssert( result == noErr, @"Couldn't update the Keychain Item with error %d.", (int)result);
+        NSDictionary *userInfo = @{@"completion_status" : (result == noErr) ? @"succeeded" : @"failed",
+                                   @"message" : @(result)};
+        [[NSNotificationCenter defaultCenter] postNotificationName:BOXRefreshTokenSaveToKeychainNotification
```
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m]: in Toolkit SDK, the keychain item could be anything, and not just refresh token. We should make setObject:forKey: to return result code. And send notification from BOXContentSDK's storeCredentialsToKeychain where we call setObject to persist session credentials including user info and tokens.

File: BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m (hunk 1 above)
Comment (jlawton) [reply BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m]: Not sure I understand the purpose of converting to a string here.

**Review by thuyBox — APPROVED** [review]

**Pull request merged**